services/app-web/src/components/ChangeHistory/ChangeHistoryV2.tsx

```diff
@@ -0,0 +1,135 @@
+import styles from './ChangeHistory.module.scss'
+type ChangeHistoryProps = {
+    contract: Contract
+}
+
+type flatRevisions = UpdateInformation & {
+    kind: 'submit' | 'unlock'
+    revisionVersion: string | undefined
+}
+
+export const ChangeHistoryV2 = ({
+    contract,
+}: ChangeHistoryProps): React.ReactElement => {
+    const flattenedRevisions = (): flatRevisions[] => {
+        const result: flatRevisions[] = []
+        //Reverse revisions to order from earliest to latest revision. This is to correctly set version for each
+        // contract & recontract.
+        const reversedRevisions = [...contract.packageSubmissions].reverse()
+        reversedRevisions.forEach((r, index) => {
+            if (r.contractRevision.unlockInfo) {
+                const newUnlock: flatRevisions = {} as flatRevisions
+                newUnlock.updatedAt = r.contractRevision.unlockInfo.updatedAt
+                newUnlock.updatedBy = r.contractRevision.unlockInfo.updatedBy
+                newUnlock.updatedReason =
+                    r.contractRevision.unlockInfo.updatedReason
+                newUnlock.kind = 'unlock'
+                //Use unshift to push the latest revision unlock info to the beginning of the array
+                result.unshift(newUnlock)
+            }
+            if (r.contractRevision.submitInfo) {
+                const newSubmit: flatRevisions = {} as flatRevisions
+
+                // Only set revisionVersion if not the latest revision and if not unlocked. This is not the same
+                // as the order of change history. The version correlates with each submitted revision.
+                const revisionVersion =
+                    index !== reversedRevisions.length - 1
+                        ? String(index + 1) //Offset version, we want to start at 1
+                        : undefined
+
+                newSubmit.updatedAt = r.contractRevision.submitInfo.updatedAt
+                newSubmit.updatedBy = r.contractRevision.submitInfo.updatedBy
+                newSubmit.updatedReason =
+                    r.contractRevision.submitInfo.updatedReason
+                newSubmit.kind = 'submit'
+                newSubmit.revisionVersion = revisionVersion
+                //Use unshift to push the latest revision submit info to the beginning of the array
+                result.unshift(newSubmit)
+            }
+        })
+        return result
+    }
+
+    const revisionHistory = flattenedRevisions()
+
+    const revisedItems: AccordionItemProps[] = revisionHistory.map(
+        (r, index) => {
+            const isInitialSubmission = r.updatedReason === 'Initial contract'
+            const isSubsequentSubmission = r.kind === 'submit'
+            // We want to know if this package has multiple contracts. To have multiple contracts, there must be minimum
+            // 3 revisions in revisionHistory the initial contract revision, unlock revision and recontract revision.
```

Comment:
nit: this sentences confuses me. A package can only have one contract. It can have multiple contract submissions in its history, it can have multiple rates, it can have multiple related submissions (via linked rates) but I dont think it can have multiple contracts

Reply:
Thanks, I could use extra eyes on this [change ](https://github.com/Enterprise-CMCS/managed-care-review/pull/2350/files#diff-1c7f359d5e6984ab5d140a1fd69d46a4490e742ab901468f6707274c00d1c244R67). I think we no longer want the cutoff to happen at >= 3?